blog/en-us/dubbo-annotation-driven.md

```diff
@@ -0,0 +1,806 @@
+
+
+
+##Annotation-Driven
+
+
+
+
+
+### `@DubboComponentScan`
+
+
+
+
+
+#### Start version： `2.5.7`
+
+
+
+
+
+
+
+#### `<dubbo:annotation> `Historical issues
+
+
+
+
+
+##### 1. Insufficient support for annotations
+
+
+
+In Dubbo 2.5.7version before, Dubbo provides two core notes `@Service`and `@Reference`, Dubbo Dubbo and service delivery service references are used.
+
+
+Among them, `@Service` as an XML element <dubbo:service>replacement annotated with Spring Framework @org.springframework.stereotype.Serviceis similar to the service provider Dubbo service exposure. Corresponding to `@Reference` it is an alternative <dubbo:referenceelement, similar to Spring `@Autowired`.
+
+
+
+`2.5.7` The previous Dubbo, similar to the earlier Spring Framework 2.5, has insufficient annotation support. The annotation needs to be used with an XML configuration file as follows:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:dubbo="http://code.alibabatech.com/schema/dubbo"
+       xmlns="http://www.springframework.org/schema/beans"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-2.5.xsd
+	http://code.alibabatech.com/schema/dubbo http://code.alibabatech.com/schema/dubbo/dubbo.xsd">
+
+    <dubbo:application name="annotation-provider"/>
+    <dubbo:registry address="127.0.0.1:4548"/>
+    <dubbo:annotation package="com.alibaba.dubbo.config.spring.annotation.provider"/>
+
+</beans>
+```
+
+
+
+
+
+##### 2.  `@Service` Bean does not support Spring AOP
+
+
+
+Meanwhile, the use of `<dubbo:annotation>`Dubbo scanned after `@Service`a problem in terms Spring agents, such as the issue GitHub [https://github.com/alibaba/dubbo/issues/794:](https://github.com/alibaba/dubbo/issues/794%EF%BC%9A)
+
+
+> About the dubbo @Service annotation when generating the ServiceBean, the interface gets the bug of the spring proxy object.
+>
+> >In the project, I used it.
+> >
+> >```java
+> >@Service
+> >@Transactional
+> >@com.alibaba.dubbo.config.annotation.Service
+> >public class SUserJpushServiceImp
+> >```
+> >
+> >In the form of to expose the service. But when publishing services, interface class is by
+> >``
+> >serviceConfig.setInterface(bean.getClass().getInterfaces()[0]);
+> >``
+> >obtaining a form, just, I have used the service @Transactional annotation, the object is a proxy. So the interface obtained is Spring's proxy interface...
+
+
+
+Many enthusiastic partners not only discovered this legacy, but also proposed some repairs. At the same time, in order to better adapt to the Spring life cycle and transition Dubbo completely to the annotation-driven programming model, a new Dubbo component scan annotation was introduced `@DubboComponentScan`.
+
+
+
+> Note: `<dubbo:annotation>` the Spring AOP issue will be `2.5.9`fixed in: [https://github.com/alibaba/dubbo/issues/1125](https://github.com/alibaba/dubbo/issues/1125)
```

Comment:
space needed after `2.5.9` and also this section need change, should be fixed in XX